src/storage-adapters/mongo/index.js

```diff
@@ -0,0 +1,91 @@
+import { MongoClient, ObjectId } from 'mongodb';
+
+const PDF_MIME_TYPE = 'application/pdf';
+
+export default class MongoAdapter {
+  constructor({ database: databaseName, collection: collectionName, connectionURI }) {
+    const client = new MongoClient(connectionURI);
+
+    this.databaseName = databaseName;
+    this.collectionName = collectionName;
+    this.client = client;
+  }
+
+  async initialize() {
+    await this.client.connect();
+    const db = this.client.db(this.databaseName);
+
+    this.collection = db.collection(this.collectionName);
+  }
+
+  async finalize() {
+    await this.client.close();
+  }
+
+  async record({ serviceId, documentType, content: passedContent, mimeType, fetchDate, isRefilter, snapshotId }) {
+    let content = passedContent;
+
+    if (mimeType == PDF_MIME_TYPE) {
+      content = passedContent.toString('utf-8'); // Serialize PDF
+    }
+
+    const previousRecord = await this.getLatestRecord(serviceId, documentType);
+
+    if (previousRecord.id && previousRecord.content == content) {
```

Comment:
`===` ?

Reply:
Not really useful but we can if you prefer